bitnami/etcd/templates/statefulset.yaml

```diff
@@ -0,0 +1,357 @@
+          value: "{{ $etcdPeerProtocol }}://0.0.0.0:{{ .Values.service.peerPort }}"
+
+        ## Clustering configuration
+        {{- if gt $replicaCount 1 }}
+        - name: ETCD_INITIAL_CLUSTER_TOKEN
+          value: "etcd-cluster-k8s"
+        - name: ETCD_INITIAL_CLUSTER_STATE
+          value: "new"
+        - name: ETCD_INITIAL_CLUSTER
+          value: {{range $i, $e := until $replicaCount }}{{ $etcdFullname }}-{{ $e }}={{ $etcdPeerProtocol }}://{{ $etcdFullname }}-{{ $e }}.{{ $etcdHeadlessServiceName }}.default.svc.cluster.local:{{ $peerPort }},{{ end }}
+        {{- end }}
+
+        ## RBAC authentication
+        - name: ALLOW_NONE_AUTHENTICATION
+          {{- if .Values.auth.rbac.enabled }}
+          value: "no"
+          {{- else if .Values.allowNoneAuthentication }}
+          value: "yes"
+          {{- else }}
+          value: "no"
+          {{- end }}
+        {{- if .Values.auth.rbac.enabled }}
+        - name: ETCD_ROOT_PASSWORD
+          valueFrom:
+            secretKeyRef:
+              name: {{ if .Values.auth.existingSecret }}{{ .Values.auth.existingSecret }}{{ else }}{{ template "etcd.fullname" . }}{{ end }}
+              key: etcd-root-password
+        {{- end }}
+
+        ## Client-to-server communication
+        {{- if and .Values.auth.client.secureTransport .Values.auth.client.useAutoTLS }}
+        - name: ETCD_AUTO_TLS
+          value: "true"
+        {{- else if .Values.auth.client.secureTransport }}
+        - name: ETCD_CERT_FILE
+          value: "/opt/bitnami/etcd/certs/client/cert.pem"
+        - name: ETCD_KEY_FILE
+          value: "/opt/bitnami/etcd/certs/client/key.pem"
+        {{- if .Values.auth.client.enableAuthentication }}
+        - name: ETCD_CLIENT_CERT_AUTH
+          value: "true"
+        - name: ETCD_TRUSTED_CA_FILE
+          value: "/opt/bitnami/etcd/certs/client/ca.crt"
+        {{- end }}
+        {{- end }}
+
+        ## Peer-to-peer communication
+        {{- if and .Values.auth.peer.secureTransport .Values.auth.peer.useAutoTLS }}
+        - name: ETCD_PEER_AUTO_TLS
+          value: "true"
+        {{- else if .Values.auth.peer.secureTransport }}
+        - name: ETCD_PEER_CERT_FILE
+          value: "/opt/bitnami/etcd/certs/peer/cert.pem"
+        - name: ETCD_PEER_KEY_FILE
+          value: "/opt/bitnami/etcd/certs/peer/key.pem"
+        {{- if .Values.auth.peer.enableAuthentication }}
+        - name: ETCD_PEER_CLIENT_CERT_AUTH
+          value: "true"
+        - name: ETCD_PEER_TRUSTED_CA_FILE
+          value: "/opt/bitnami/etcd/certs/peer/ca.crt"
+        {{- end }}
+        {{- end }}
+
+        ports:
+        - name: client
+          containerPort: {{ .Values.service.port }}
+        - name: peer
+          containerPort: {{ .Values.service.port }}
+        lifecycle:
+          preStop:
+            exec:
+              command:
+                - "/bin/sh"
+                - "-ec"
+                - |
+                  AUTH_OPTIONS=""
```

Comment:
After seeing this amount of bash code in the yaml, would it make sense to move it as a template?

Reply:
it makes sense yes. I also thought about that as an option but I choose the bash path as it was more straightforward for me. I will try to create a template in any case